docs/source/developers/java/building.rst

```diff
@@ -171,11 +171,30 @@ IDE Configuration
 IntelliJ
 --------
 
-To start working on Arrow in IntelliJ, just open the `java/`
-subdirectory of the Arrow repository.
+To start working on Arrow in IntelliJ: build the project once from the command
+line using ``mvn clean install``. Then open the ``java/`` subdirectory of the
+Arrow repository, and update the following settings:
 
+* In the Files tool window, find the path ``vector/target/generated-sources``,
+  right click the directory, and select Mark Directory as > Generated Sources
+  Root.
```

Comment:
Does it make sense to say that there's no need to mark the other generated-sourced folders as Generated Sources Root? There are, I think, five folders (under tools, plasma, format, compression, and algorithm) that are generated, but they contain only empty "annotation" folders, at least in my current build.